src/libs/ValidationUtils.ts

```diff
@@ -503,6 +503,33 @@ function isValidSubscriptionSize(subscriptionSize: string): boolean {
     return !Number.isNaN(parsedSubscriptionSize) && parsedSubscriptionSize > 0 && parsedSubscriptionSize <= CONST.SUBSCRIPTION_SIZE_LIMIT && Number.isInteger(parsedSubscriptionSize);
 }
 
+/**
+ * Validates the given value if it is correct email address.
+ * @param email
+ */
+function isValidEmail(email: string): boolean {
+    return Str.isValidEmail(email);
+}
+
+/**
+ * Validates the given value if it is correct phone number in E164 format (international standard).
+ * @param phoneNumber
+ */
+function isValidPhoneInternational(phoneNumber: string): boolean {
+    const phoneNumberWithCountryCode = LoginUtils.appendCountryCode(phoneNumber);
+    const parsedPhoneNumber = parsePhoneNumber(phoneNumberWithCountryCode);
+
+    return Str.isValidE164Phone(parsedPhoneNumber.number?.e164 ?? '');
```

Comment:
```suggestion
    return parsedPhoneNumber.possible && Str.isValidE164Phone(parsedPhoneNumber.number?.e164 ?? '');
```

Reply:
Let's use add `parsedPhoneNumber.possible` into the condition. `isValidE164Phone` uses [this regex](https://github.com/Expensify/expensify-common/blob/2f96348d48fb9657b95aae12567e6846e199da88/lib/CONST.ts#L629), which is not enough.

<img width="1020" alt="Screenshot 2024-11-06 at 17 36 14" src="https://github.com/user-attachments/assets/16ef9546-860a-44d7-92e3-1bef7f82ee79">

Reply:
Updated

Reply:
All good! I'll work on the checklist!